Un-silence a unit test: test_rgb.py

### Context

While working on https://github.com/ivadomed/ivadomed/pull/714, I had to silence the function `save_color_labels` because the way it was implemented produced an error when the header of the reference image was copied to the output prediction. Namely, the field "dtype" was not compatible, yielding the following error when saving the  NIFTI file:
~~~
>           raise WriterError('Cannot cast to or from non-numeric types')
E           nibabel.arraywriters.WriterError: Cannot cast to or from non-numeric types
~~~

### TODO

- Fix the implementation of `save_color_labels`
- Reestablish the calls to this function, which were removed in https://github.com/ivadomed/ivadomed/pull/714/commits/52079bd339b6e0cdffa61b778da2bae9a3560e55
- I had to silence a test (see this commit https://github.com/ivadomed/ivadomed/pull/714/commits/0f946fc3fe4b083d33783fe96a4be5e9ed207558), which was lower priority compared to the PR. This silenced test should be revisited.

--- a/ivadomed/scripts/download_data.py
+++ b/ivadomed/scripts/download_data.py
@@ -22,7 +22,7 @@ DICT_URL = {
             `Spine Generic <https://github.com/spine-generic/data-multi-subject>`_.
             Used for Tutorial and example in Ivadomed."""},
     "data_testing": {
-        "url": ["https://github.com/ivadomed/data-testing/archive/r20210322.zip"],
+        "url": ["https://github.com/ivadomed/data-testing/archive/r20210628.zip"],
         "description": "Data Used for integration/unit test in Ivadomed."},
     "t2_tumor": {
         "url": ["https://github.com/ivadomed/t2_tumor/archive/r20200621.zip"],

--- a/testing/unit_tests/test_rgb.py
+++ b/testing/unit_tests/test_rgb.py
@@ -2,7 +2,7 @@ from ivadomed import visualize as imed_visualize
 import numpy as np
 import torch
 import os
-from testing.unit_tests.t_utils import create_tmp_dir,  __data_testing_dir__, __tmp_dir__
+from testing.unit_tests.t_utils import create_tmp_dir,  __data_testing_dir__, __tmp_dir__, download_data_testing_test_files
 from testing.common_testing_util import remove_tmp_dir
 
 def setup_function():
@@ -12,23 +12,23 @@ def setup_function():
 # TODO (#720): un-silence this test. This test was silenced because it was failing after introducing the copy of the header in #714.
 #  It might be possible to solve the issue with the test by initializing the "image" with a proper shape, i.e. one that
 #  would match that of the prediction.
-# def test_save_rgb():
-#     # Create image with shape HxWxDxC
-#     image = [[[[0 for i in range(10)] for i in range(10)] for i in range(4)] for i in range(3)]
-#     for i in range(3):
-#         image[0][2][5][i] = 1
-#         image[1][2][5][i+3] = 1
-#         image[2][2][5][i + 6] = 1
-#     image_n = np.array(image)
-#     imed_visualize.save_color_labels(
-#         gt_data=image_n,
-#         binarize=False,
-#         gt_filename=os.path.join(
-#             __data_testing_dir__,
-#             "derivatives/labels/sub-unf01/anat/sub-unf01_T2w_lesion-manual.nii.gz"),
-#         output_filename=os.path.join(__tmp_dir__, "rgb_test.nii.gz"),
-#         slice_axis=0
-#     )
+def test_save_rgb(download_data_testing_test_files):
+    # Create image with shape HxWxDxC
+    image = [[[[0 for i in range(10)] for i in range(10)] for i in range(4)] for i in range(3)]
+    for i in range(3):
+        image[0][2][5][i] = 1
+        image[1][2][5][i+3] = 1
+        image[2][2][5][i + 6] = 1
+    image_n = np.array(image)
+    imed_visualize.save_color_labels(
+        gt_data=image_n,
+        binarize=False,
+        gt_filename=os.path.join(
+            __data_testing_dir__,
+            "rgb_test_file.nii.gz"),
+        output_filename=os.path.join(__tmp_dir__, "rgb_test.nii.gz"),
+        slice_axis=0
+    )
 
 
 def test_rgb_conversion():

--- a/testing/unit_tests/test_rgb.py
+++ b/testing/unit_tests/test_rgb.py
@@ -9,9 +9,6 @@ def setup_function():
     create_tmp_dir()
 
 
-# TODO (#720): un-silence this test. This test was silenced because it was failing after introducing the copy of the header in #714.
-#  It might be possible to solve the issue with the test by initializing the "image" with a proper shape, i.e. one that
-#  would match that of the prediction.
 def test_save_rgb(download_data_testing_test_files):
     # Create image with shape HxWxDxC
     image = [[[[0 for i in range(10)] for i in range(10)] for i in range(4)] for i in range(3)]

